File: Alay/Sliding-Window/1423.cpp
class Solution {
public:
    int maxScore(std::vector<int>& cardPoints, int k) {
        int n = cardPoints.size();
        int totalSum = 0;
        for (int i = 0; i < n; i++) {
            totalSum += cardPoints[i];
        }
        if (k == n) {
            return totalSum;
        }
        int windowSum = 0, minWindowSum = INT_MAX;
        for (int i = 0; i < n; i++) {
            windowSum += cardPoints[i];
            if (i >= n - k - 1) {
                minWindowSum = min(minWindowSum, windowSum);
                windowSum -= cardPoints[i - (n - k - 1)];
            }
        }
        return totalSum - minWindowSum;
    }
};

File: Aditya-Pratap/Sliding Window/1423.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        if len(cardPoints)==k:
            return sum(cardPoints)

        checkSum=sum(cardPoints[:k])
        maxSum=checkSum
        j=len(cardPoints)-1
        i=k-1
        while i>-1:
            checkSum += cardPoints[j]-cardPoints[i]
            i -=1
            j -=1
            maxSum=max(checkSum,maxSum)
        return maxSum
